robot_sort/robot_sort.py

```diff
@@ -97,8 +97,33 @@ def sort(self):
         Sort the robot's list.
         """
         # Fill this out
-        pass
+        # Implement bubble sort
+        # Use light to keep track of boolean
+        self.set_light_on()
 
+        while self.light_is_on():
+            self.set_light_off()
+
+        # Loop through list using can_move_right()
+            while self.can_move_right():
+                # Pick up item and move to next place
+                self.swap_item()
+                self.move_right()
+
+                # If holding greater value, swap, turn on light and move smaller item back a place
+                if self.compare_item() == 1:
+                    self.swap_item()
+                    self.set_light_on()
+                    self.move_left()
+                    self.swap_item()
+                else:
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+
+            # At end of list with largest number in correct place, move back to start
+            while self.can_move_left():
```

Comment:
You could optimize the sort here by "bubbling" the smallest item to the left just like how you are bubbling the largest item to the right.